tests/e2e/k8s_helpers_test.go

```diff
@@ -0,0 +1,172 @@
+	t.Helper()
+
+	slackEnabledEnvName := appCfg.Deployment.Envs.SlackEnabledName
+	slackChannelIDEnvName := appCfg.Deployment.Envs.SlackChannelIDName
+
+	deployment, err := deployNsCli.Get(context.Background(), appCfg.Deployment.Name, metav1.GetOptions{})
+	require.NoError(t, err)
+	require.NotNil(t, deployment)
+
+	containerIdx, exists := findContainerIdxByName(deployment, appCfg.Deployment.ContainerName)
+	require.True(t, exists)
+
+	envs := deployment.Spec.Template.Spec.Containers[containerIdx].Env
+
+	var originalEnvs []v1.EnvVar
+	channelIDEnv, exists := findEnv(envs, slackChannelIDEnvName)
+	if exists {
+		originalEnvs = append(originalEnvs, channelIDEnv)
+	}
+	enabledEnv, exists := findEnv(envs, slackEnabledEnvName)
+	if exists {
+		originalEnvs = append(originalEnvs, enabledEnv)
+	}
+
+	restoreDeployEnvsFn := func(t *testing.T) {
+		t.Helper()
+		t.Logf("Restoring envs for deployment...")
+		deployment, err := deployNsCli.Get(context.Background(), appCfg.Deployment.Name, metav1.GetOptions{})
+		require.NoError(t, err)
+		require.NotNil(t, deployment)
+
+		containerIdx, exists := findContainerIdxByName(deployment, appCfg.Deployment.ContainerName)
+		require.True(t, exists)
+
+		deployment.Spec.Template.Spec.Containers[containerIdx].Env = updateEnv(envs,
+			nil,
+			[]string{slackEnabledEnvName, slackChannelIDEnvName},
+		)
+
+		if len(originalEnvs) > 0 {
+			deployment.Spec.Template.Spec.Containers[containerIdx].Env = updateEnv(envs, originalEnvs, nil)
+		}
+		_, err = deployNsCli.Update(context.Background(), deployment, metav1.UpdateOptions{})
+		require.NoError(t, err)
+	}
+
+	deployment.Spec.Template.Spec.Containers[containerIdx].Env = updateEnv(
+		envs,
+		[]v1.EnvVar{
+			{Name: slackEnabledEnvName, Value: strconv.FormatBool(true)},
+			{Name: slackChannelIDEnvName, Value: channelName},
+		},
+		nil,
+	)
+
+	_, err = deployNsCli.Update(context.Background(), deployment, metav1.UpdateOptions{})
+	require.NoError(t, err)
+
+	return restoreDeployEnvsFn
+}
+
+func waitForDeploymentReady(deployNsCli appsv1cli.DeploymentInterface, deploymentName string, waitTimeout time.Duration) error {
+	// Wait for deployment ready
+	var lastErr error
+	err := wait.Poll(pollInterval, waitTimeout, func() (done bool, err error) {
+		deployment, err := deployNsCli.Get(context.Background(), deploymentName, metav1.GetOptions{})
+		if err != nil {
+			lastErr = err
+			return false, nil
+		}
+
+		condition := deploymentutil.GetDeploymentCondition(deployment.Status, appsv1.DeploymentAvailable)
+		if condition == nil {
+			lastErr = fmt.Errorf("deployment condition %q is nil", appsv1.DeploymentAvailable)
+			return false, nil
+		}
+
+		return condition.Status == v1.ConditionTrue, nil
+	})
+	if err != nil {
+		if err == wait.ErrWaitTimeout {
+			return lastErr
+		}
+		return err
+	}
+
+	return nil
+}
+
+func findContainerIdxByName(deployment *appsv1.Deployment, containerName string) (int, bool) {
```

Comment:
IMO it can be simplified to:
```go
func findContainerIdxByName(deployment *appsv1.Deployment, containerName string) int {
	const notFound = -1
	if deployment == nil {
		return notFound
	}

	containers := deployment.Spec.Template.Spec.Containers
	for i, container := range containers {
		if container.Name == containerName {
			return i
		}
	}

	return notFound
}
```
or if you still want to have bool indicator:

```go
func findContainerIdxByName(deployment *appsv1.Deployment, containerName string) (int, bool) {
	notFound := func() (int, bool) {
		return -1, false
	}
	if deployment == nil {
		return notFound()
	}

	containers := deployment.Spec.Template.Spec.Containers
	for i, container := range containers {
		if container.Name == containerName {
			return i, true
		}
	}

	return notFound()
}
// or
func findContainerIdxByName(deployment *appsv1.Deployment, containerName string) (int, bool) {
	if deployment == nil {
		return -1, false
	}

	containers := deployment.Spec.Template.Spec.Containers
	for i, container := range containers {
		if container.Name == containerName {
			return i, true
		}
	}

	return -1, false
}
```